dev/breeze/src/airflow_breeze/commands/release_management_commands.py

```diff
@@ -949,6 +950,72 @@ def run_generate_constraints_in_parallel(
     )
 
 
+@release_management.command(
+    name="tag-providers",
+    help="Generates tags for airflow provider releases.",
+)
+@option_dry_run
+@option_verbose
+def tag_providers():
+    found_remote = None
+    remotes = ["origin", "apache"]
+    for remote in remotes:
+        try:
+            command = ["git", "remote", "get-url", "--push", shlex.quote(remote)]
+            result = subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.DEVNULL, text=True)
+            if "apache/airflow.git" in result.stdout:
+                found_remote = remote
+                break
+        except subprocess.CalledProcessError:
+            pass
+
+    if found_remote is None:
+        raise ValueError("Could not find remote configured to push to apache/airflow")
+
+    tags = []
+    for file in os.listdir(os.path.join(SOURCE_DIR_PATH, "dist")):
+        if file.endswith(".whl"):
+            match = re.match(r".*airflow_providers_(.*)-(.*)-py3.*", file)
+            if match:
+                provider = f"providers-{match.group(1).replace('_', '-')}"
+                tag = f"{provider}/{match.group(2)}"
+                try:
+                    subprocess.run(
+                        ["git", "tag", shlex.quote(tag), "-m", f"Release {date.today()} of providers"],
+                        check=True,
+                    )
+                    tags.append(tag)
+                except subprocess.CalledProcessError:
+                    pass
+
+    if tags and len(tags) > 0:
+        try:
+            push_command = ["git", "push", remote] + [shlex.quote(tag) for tag in tags]
+            push_result = subprocess.Popen(
+                push_command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True
+            )
+            push_output, push_error = push_result.communicate()
+            if push_output:
+                get_console().print("[success]{push_output}[/]")
+            if push_error:
+                get_console().print("[error]{push_error}[/]")
+            get_console().print("\n[success]Tags pushed successfully.[/]")
+        except subprocess.CalledProcessError:
+            get_console().print("\n[error]Failed to push tags, probably a connectivity issue to Github.[/]")
+            clean_local_tags = os.environ.get("CLEAN_LOCAL_TAGS", "true").lower() == "true"
```

Comment:
This one should be turned into a new `--clean-local-tags` click flag with `envvar="CLEAN_LOCAL_TAGS"` - this way it will be discoverable in `--help` and you can add nice description there, at the same time it will read from the envvar automatically if specified, so you do not have to read it manually from os.ENVIRONMENT. You can set/show default in such flag etc.

Reply:
Added this flag